### PR TITLE
P11 Dev: Gtest cases for DDIMM parser

### PR DESCRIPTION
**Pull request by branupama**
Adding different google unit test cases to verify DDIMM parser is working or not with different inputs.

Test:
Observed all google test cases got passed with different inputs to DDIMM parser.

### CODE REVIEW

**Comment by PriyangaRamasamy** [comment]
jenkins run tests please

**Comment by geissonator** [comment]
add to approvelist

File: include/constants.hpp [hunk 1]
```diff
@@ -1,6 +1,9 @@
 #pragma once
 
+#include <stdint.h>
```
Comment (SunnySrivastava1984) [inline include/constants.hpp:3]: Why do you need this here?

File: test/utest_ddimm_parser.cpp [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+    types::BinaryVector l_emptyVector{};
+    DdimmVpdParser l_ddimmParser(l_emptyVector);
+
+    EXPECT_THROW(l_ddimmParser.parse(), std::exception);
+}
+
+TEST_F(DdimmVpdParserTest, InvalidVpdType)
+{
+    // Invalid VPD type - corrupted at index[3]
+    m_ddimmVpdVector[constants::SPD_BYTE_3] = constants::SPD_MODULE_TYPE_DDIMM +
```
Comment (SunnySrivastava1984) [inline test/utest_ddimm_parser.cpp:96]: What is the point of having the file is vector that is being passed is hardcoded?

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: include/constants.hpp (hunk 1 above)
Comment (branupama) [reply include/constants.hpp:3]: added to resolve the error - unknown type name 'uint8_t'
Got above error when I included constants.hpp in the utest*.cpp file.

Comment (SunnySrivastava1984) [reply include/constants.hpp:3]: But why would this be needed here? Could you please check if the requirement us in constant.hpp or in the test file.

File: test/utest_ddimm_parser.cpp (hunk 2 above)
Comment (branupama) [reply test/utest_ddimm_parser.cpp]: Updated the test case.

File: test/utest_ddimm_parser.cpp [hunk 3]
```diff
@@ -0,0 +1,86 @@
...
+    // Badly formed DDIMM VPD data - corrupted at index[235],
+    // ddimm size calculated a zero
+    nlohmann::json l_json;
+    std::string l_vpdFile("vpd_files/ddr5_ddimm_corrupted_index_235.dat");
+    Parser l_vpdParser(l_vpdFile, l_json);
+
+    EXPECT_THROW(l_vpdParser.parse(), std::exception);
+}
+
+TEST(DdimmVpdParserTest, InvalidValue)
```
Comment (SunnySrivastava1984) [inline test/utest_ddimm_parser.cpp:61]: Use better naming so that it can be directly mapped to what is being checked.

**Review by SunnySrivastava1984 — CHANGES_REQUESTED** [review]

File: test/utest_ddimm_parser.cpp (hunk 3 above)
Comment (branupama) [reply test/utest_ddimm_parser.cpp]: Done

**Review by SunnySrivastava1984 — APPROVED** [review]

**Pull request merged**